### DISCUSSION TITLE
Question about Euphony build

### DISCUSSION THREAD

**Discussion by kldaji** [euphony-io/euphony#146] [Q&A] [answered]
Hi, I got a problem with Euphony build.

When I build Euphony project, I got an error message like below.

![스크린샷 2022-07-14 오후 11 25 58](https://user-images.githubusercontent.com/78070388/179006179-d060dc8e-df15-4895-a4ed-6a9087be8ad8.png)

But, I found that I have `euphony/.cxx/cmake/debug/arm64-v8a/android_gradle_build_stdout_targets.txt` file like below.

![스크린샷 2022-07-14 오후 11 25 33](https://user-images.githubusercontent.com/78070388/179006113-c51e89de-f763-4bd5-b5bc-b32a27dfffe5.png)

How can I solve this problem?

**Comment by designe** [comment, marked as answer]
Hi! @kldaji this is the known issue!

Actually, this issue occurs after clean build.
After clean build, there are no abi log file. so, it will be occured.

Could you change for loop code(60 lines) in your build.gradle?

```gradle
for (abi in abiArr) {
 def filePath = "euphony/.cxx/cmake/debug/$abi/$logfile"
 def file = new File(filePath)
 if (file.exists()) {
  def lines = file.readLines()
  if (lines.get(lines.size() - 1).contains(lastWordIfThereAreNoDevice))
   continue
  noDevice = false
  resultStr += "> $abi/$logfile\n"
  resultStr += lines.join("\n")
  }
}
```

It hasn't been reflected in the master branch yet.
please follow this link https://github.com/designe/euphony/commit/4bec7c376052c466a59c9924cb2d492051b70851

Let me know your build status after modify code like above :)

**Reply by kldaji** [reply]
Thanks!!
It totally works!! 👍👍
